go-controller/pkg/ovn/external_ids_syncer/acl/acl_sync_test.go

```diff
@@ -127,7 +127,24 @@ var _ = ginkgo.Describe("OVN ACL Syncer", func() {
 				),
 			},
 		}
-		testSyncerWithData(testData, controllerName, []libovsdbtest.TestData{}, nil, nil)
+		initialDb := []libovsdbtest.TestData{
+			&nbdb.PortGroup{
+				UUID: types.ClusterPortGroupNameBase,
+				ACLs: []string{
+					testData[0].after.String() + "-UUID",
+					testData[1].before.Match,
+				},
+			},
+		}
+		finalDb := []libovsdbtest.TestData{
+			&nbdb.PortGroup{
+				UUID: types.ClusterPortGroupNameBase,
+				ACLs: []string{
+					testData[0].after.String() + "-UUID",
+				},
+			},
+		}
```

Comment:
I think with some minor changes to `testSyncerWithData` you won't need test tweaks with initial and final DB, something like:
```
func testSyncerWithData(data []aclSync, controllerName string, initialDbState, finalDbState []libovsdbtest.TestData,
	existingNodes []*v1.Node) {
	// create initial db setup
	pgBefore := &nbdb.PortGroup{
		UUID: types.ClusterPortGroupNameBase,
	}
	dbSetup := libovsdbtest.TestSetup{NBData: append(initialDbState, pgBefore)}
	for _, asSync := range data {
		if asSync.after != nil {
			asSync.before.UUID = asSync.after.String() + "-UUID"
		} else {
			asSync.before.UUID = asSync.before.Match
		}
		pgBefore.ACLs = append(pgBefore.ACLs, asSync.before.UUID)
		dbSetup.NBData = append(dbSetup.NBData, asSync.before)
	}
	libovsdbOvnNBClient, _, libovsdbCleanup, err := libovsdbtest.NewNBSBTestHarness(dbSetup)
	defer libovsdbCleanup.Cleanup()
	gomega.Expect(err).NotTo(gomega.HaveOccurred())
	// create expected data using addressSetFactory
	var expectedDbState []libovsdbtest.TestData
	if finalDbState != nil {
		expectedDbState = finalDbState
	} else {
		expectedDbState = initialDbState
	}
	pgAfter := &nbdb.PortGroup{
		UUID: types.ClusterPortGroupNameBase,
	}
	expectedDbState = append(expectedDbState, pgAfter)
	for _, aclSync := range data {
		if aclSync.after != nil {
			acl := aclSync.before.DeepCopy()
			acl.ExternalIDs = aclSync.after.GetExternalIDs()
			acl.Tier = types.DefaultACLTier
			pgAfter.ACLs = append(pgAfter.ACLs, acl.UUID)
			expectedDbState = append(expectedDbState, acl)
		}
	}
	// run sync
	syncer := NewACLSyncer(libovsdbOvnNBClient, controllerName)
	err = syncer.SyncACLs(existingNodes)
	gomega.Expect(err).NotTo(gomega.HaveOccurred())
	// check results
	gomega.Eventually(libovsdbOvnNBClient).Should(libovsdbtest.HaveData(expectedDbState))
}
```